webui/src/pages/repositories/repository/error.jsx

```diff
@@ -0,0 +1,37 @@
+import {useRouter} from "../../../lib/hooks/router";
+import Alert from "react-bootstrap/Alert";
+import Button from "react-bootstrap/Button";
+import {repositories, RepositoryDeletionError} from "../../../lib/api";
+import {TrashIcon} from "@primer/octicons-react";
+import React from "react";
+import {Error} from "../../../lib/components/controls";
+
+const RepositoryInDeletionContainer = (repoId) => {
+    const router = useRouter();
+    return (
+        <Alert variant="warning">
+            <Alert.Heading>Repository is undergoing deletion</Alert.Heading>
+            This may take several seconds. You can retry the deletion process by pressing the delete button again.
+            <hr />
+            <div className="d-flex justify-content-end">
+                <Button variant="danger" className="mt-3" onClick={ async () => {
+                    try {
+                        await repositories.delete(repoId);
+                    } catch {
+                        // continue regardless of error
+                    }
+                    return router.push('/repositories')
+                }}>
+                    <TrashIcon/> Delete Repository
+                </Button>
+            </div>
+        </Alert>
+    );
+};
+
+export const RepoError = ({error}) => {
```

Comment:
Nice one!

Reply:
However it might be hard for the next developer to understand when to use `Error` and when to use `RepoError`. 
Consider switching the usage, let everyone call the simple `Error` and there perform this check and initiate `RepositoryInDeletionContainer` for `RepositoryDeletionError`.

Reply:
We wanted to bound the usage of RepoError to the repository modules so that it will be accessible only in this context. Also this error is being used only on specific flows